usr/libexec/greenboot/greenboot

```diff
@@ -19,13 +20,17 @@ script_runner () {
           echo "<2>$failure_msg. Continuing..." >&2
           ;;
         "strict")
-          echo "<0>$failure_msg" >&2
-          exit $rc
+          required_hc_failed=true
+          echo "<0>$failure_msg. Continuing..." >&2
       esac
     else
       echo "<6>Script '$(basename "$script")' SUCCESS"
     fi
   done
+
+  if [[ $required_hc_failed == true ]]; then
+    exit 1
```

Comment:
maybe echo something here, too, noting some (at least one) tests have failed?

Reply:
Good idea.

Reply:
What do you think @LorbusChris?

journalctl:
```
Oct 05 08:26:14 fedora 00_required_scripts_start.sh[681]: Running greenboot Required Health Check Scripts
Oct 05 08:26:14 fedora greenboot[670]: Script '00_required_scripts_start.sh' SUCCESS
Oct 05 08:26:14 fedora greenboot[670]: Script '01_fail.sh' FAILURE (exit code '1'). Continuing...
Oct 05 08:26:15 fedora greenboot[670]: Script '02_fail.sh' FAILURE (exit code '1'). Continuing...
Oct 05 08:26:15 fedora zsh.sh[694]: ZSH is available
Oct 05 08:26:15 fedora greenboot[670]: Script 'zsh.sh' SUCCESS
Oct 05 08:26:15 fedora greenboot[670]: ATTENTION: At least one required health check has failed
```

MOTD:
```
Script '01_fail.sh' FAILURE (exit code '1'). Continuing...
Script '02_fail.sh' FAILURE (exit code '1'). Continuing...
ATTENTION: At least one required health check has failed
Boot Status is RED - Health Check FAILURE!
```

If OK, will commit and update the PR.

Reply:
LGTM, although attention would sound like a warning to me, not an error. Not too important here though I think.

Reply:
Yeah, I wasn't too sure about that choice of words either but couldn't think of a better one. @nullr0ute, any feedback?

Reply:
We should possibly also report which of the health checks failed so the user knows where to look.

Reply:
That should already be the case here (e.g. `Script '02_fail.sh' FAILURE (exit code '1'). Continuing...` in the MOTD above)